src/rust/ide/src/controller/graph.rs

```diff
@@ -68,6 +68,8 @@ pub struct NoPatternOnNode {
 pub enum Notification {
     /// The content should be fully reloaded.
     Invalidate,
+    /// The graph expressions need updating, e.g., types, names, ports.
+    ExpressionUpdate,
```

Comment:
I'm unsure of this name. The expression means just the same as "node label" - the code. At least in controllers. But this event is actually fired not when code changes, but type or span-tree.

Therefore I would rename it to the `PortsUpdate`